meta-mender-core/recipes-core/base-files/base-files_3.%.bbappend

```diff
@@ -8,13 +8,18 @@ do_patch_device() {
     true
 }
 do_patch_device_menderimage() {
-    if [ -z "${MENDER_BOOT_PART}" ] || [ -z "${MENDER_DATA_PART}" ]; then
-        bberror "MENDER_BOOT_PART or MENDER_DATA_PART not set."
+    if [ -z "${MENDER_DATA_PART}" ]; then
+        bberror "MENDER_DATA_PART not set."
         exit 1
     fi
 
-    sed -i -e 's,@MENDER_BOOT_PART@,${MENDER_BOOT_PART},g' ${S}/fstab
-    sed -i -e 's,@MENDER_BOOT_FSTYPE@,${MENDER_BOOT_FSTYPE},g' ${S}/fstab
+    if [ -n "${MENDER_BOOT_PART}" ]; then
+        sed -i -e 's,@MENDER_BOOT_PART@,${MENDER_BOOT_PART},g' ${S}/fstab
+        sed -i -e 's,@MENDER_BOOT_FSTYPE@,${MENDER_BOOT_FSTYPE},g' ${S}/fstab
+    else
+        bbwarn "MENDER_BOOT_PART not set. Removing from fstab..."
```

Comment:
Agree that `bbwarn` is a bit overkill.

Maybe drop it completely?  But might be good to have something in the logs, will update with `bbdebug`